server/api-util/idToken.js

```diff
@@ -0,0 +1,82 @@
+const crypto = require('crypto');
+const { default: fromKeyLike } = require('jose/jwk/from_key_like');
+const { default: SignJWT } = require('jose/jwt/sign');
+
+const radix = 10;
+const PORT = parseInt(process.env.REACT_APP_DEV_API_SERVER_PORT, radix);
+const rootUrl = process.env.REACT_APP_CANONICAL_ROOT_URL;
+const useDevApiServer = process.env.NODE_ENV === 'development' && !!PORT;
+
+const issuerUrl = useDevApiServer ? `http://localhost:${PORT}/api` : `${rootUrl}/api`;
+
+const clientId = process.env.CUSTOM_OIDC_CLIENT_ID;
+
+// These keys are used for signing the ID token (JWT)
+// When you store them to environment variables you should replace
+// any line brakes with '\n'.
+// You should also make sure that the key size is big enough.
+const rsaSecretKey = process.env.RSA_SECRET_KEY;
+const rsaPublicKey = process.env.RSA_PUBLIC_KEY;
```

Comment:
Similarly, as with CUSTOM_OIDC_CLIENT_ID, I think that these implementation specific keys/environment variables are a bit too deeply placed in the util file. How about, defining these in apiRouter? Something like this:

```js
const rsaSecretKey = process.env.RSA_SECRET_KEY;
const rsaPublicKey = process.env.RSA_PUBLIC_KEY;
router.get('/.well-known/jwks.json', jwksUri(rsaSecretKey, rsaPublicKey));
```

That way customizers wouldn't need to pay too much attention to this utility file, but only configure new routes in apiRouter.